packages/compose/README.md

```diff
@@ -121,7 +121,20 @@ _Returns_
 
 <a name="withGlobalEvents" href="#withGlobalEvents">#</a> **withGlobalEvents**
 
-Undocumented declaration.
+Higher-order component creator which, given an object of DOM event types and
+values corresponding to a callback function name on the component, will
+create or update a window event handler to invoke the callback when an event
+occurs. On behalf of the consuming developer, the higher-order component
+manages unbinding when the component unmounts, and binding at most a single
+event handler for the entire application.
```

Comment:
Interesting. Isn't that a built in feature of `addEventListener` to bind once at most?

> The event listener list will not contain multiple event listeners with equal type, callback, and capture, as add an event listener prevents that.
>
> https://dom.spec.whatwg.org/#dom-eventtarget-addeventlistener

Reply:
The key words there being "equal [..] callback". Since `function() {} !== function() {}`, it's not reused.

Try in the console, for example:

```
[ ...Array( 5 ) ].forEach( ( v, i ) => document.addEventListener( 'click', () => console.log( 'click' + i ) ) )
```

![image](https://user-images.githubusercontent.com/1779930/56758527-ddf93c80-6764-11e9-905f-471527791027.png)

Reply:
>Interesting. Isn't that a built in feature of addEventListener to bind once at most?

While it's not really relevant for change here, the question does get me wondering whether it really matters to consolidate to a single event handler. I'm trying to recall from the original implementation (#5730) whether I'd thought this was some sort of performance enhancement (i.e. one `'scroll'` handler on the `window` rather than one for every component) and if that was the case, whether there's actually any truth to the consolidation being an optimization. I might expect a browser with its low-level language code could iterate to invoke individual callbacks of `addEventListener` faster than we could iterate to invoke the `Listener` callbacks here in JavaScript.